dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionProcessor.java

```diff
@@ -135,6 +135,8 @@ static final protected Object getSyncObject(String name) {
     protected JspClassloaderContext jspClassloaderContext = null;
     protected JspCompilerFactory jspCompilerFactory = null;
     protected IServletContextExtended webapp = null;
+    
+    protected String loadedJSPVersion = null;
```

Comment:
This looks like it could be a  `final` field?

Reply:
I see no harm in that. When a new JSP feature is loaded, a new com.ibm.ws.jsp.webcontainerext.JSPExtensionProcessor is created with a newer version associated with it.  Let me update the code .